src/Contract/Transaction.purs

```diff
@@ -0,0 +1,153 @@
+-- | A module that defines the different transaction data types, balancing
+-- | functionality, signing and submission.
+module Contract.Transaction
+  ( balanceTx
+  , balanceTxM
+  , signTransaction
+  , submitTransaction
+  , module BalanceTxError
+  , module JsonWsp
+  , module ScriptLookups
+  , module Transaction
+  , module TxOutput
+  , module UnbalancedTx
+  ) where
+
+import Prelude
+import BalanceTx (balanceTx) as BalanceTx
+import BalanceTx (BalanceTxError) as BalanceTxError
+import Contract.Monad (Contract)
+import Data.Either (Either, hush)
+import Data.Maybe (Maybe)
+import Data.Newtype (wrap)
+import QueryM (signTransaction, submitTransaction) as QueryM
+import Types.JsonWsp (OgmiosTxOut, OgmiosTxOutRef) as JsonWsp --  Perhaps it's best not to expose `JsonWsp`.
```

Comment:
I agree with the comment

Reply:
Hmm, actually, because I followed the chain index convention, `unspentOutputs` is currently using `Map TxOutRef OgmiosTxOut` as input so we'd need to expose this for now. I did think about changing this recently, generally script lookups will need refining as mentioned before.

Reply:
https://github.com/Plutonomicon/cardano-browser-tx/issues/200